blog: add dec monthly report (#1838)
iresty/iresty.github.io: blog/en/blog/2024/12/31/monthly-report.md

---

> We have recently made some additions and improvements to specific features within Apache APISIX. The main improvements include the addition of supporting configuring response headers when using the `limit-count` plugin and supporting "system" `ssl_trusted_certificate`, among other enhancements. For detailed information, please read the monthly report.
<!--truncate-->
## Introduction

From its inception, the Apache APISIX project has embraced the ethos of open-source community collaboration, propelling it into the ranks of the most active global open-source API gateway projects. The proverbial wisdom of 'teamwork makes the dream work' rings true in our way and is made possible by the collective effort of our community.

From December 1 to December 31, 11 contributors made 27 commits to Apache APISIX. We sincerely appreciate your contributions to Apache APISIX.

## Contributor Statistics

![Apache APISIX Contributors List](https://static.apiseven.com/uploads/2024/12/31/npwFTjZH_dec-monthly-report-en.png)

![Apache APISIX New Contributors](https://static.apiseven.com/uploads/2024/12/31/LWGfMhY3_dec-new-contributors.jpg)

## Recent Feature Highlights

> 1. [Make Rate-Limit Response Headers Configurable in Limit-Count Plugin](https://github.com/apache/apisix/pull/11831) (Contributor: [shreemaan-abhishek](https://github.com/shreemaan-abhishek))

Reviewer: plugin name format?

e.g. `limit-count`

> This PR allows users to configure the names of rate-limit response headers using `plugin-metadata` when using the `limit-count` plugin.

Reviewer: ditto

Author: `limit-count` already, you mean `rate-limit`?

Reviewer: if its an adjective then it should be `rate-limiting response headers`

> This PR allows users to configure the names of rate-limiting response headers using `plugin-metadata` when using the `limit-count` plugin.

Reviewer: I don't think "plugin metadata" needs to be in inline code like this `plugin-metadata` . It's a core resource, not a plugin.

2. [Refactor Workflow Plugin Registration](https://github.com/apache/apisix/pull/11832) (Contributor: [Revolyssup](https://github.com/Revolyssup))

> Current logic requires the table `supported_actions` defined in `workflow.lua` to be changed when any new plugin is added to the workflow plugin. With this change, the registration in the table is offloaded to the newly added plugin by implementing the function `workflow_handler()`.

Reviewer: ditto

Author: workflow plugin --> `workflow` plugin

> Current logic requires the table `supported_actions` defined in `workflow.lua` to be changed when any new plugin is added to the `workflow` plugin. With this change, the registration in the table is offloaded to the newly added plugin by implementing the function `workflow_handler()`.

Reviewer: This is a refactoring; the feature has not changed at all. However the sectional title is `Recent Feature Highlights`, so it's not very appropriate.

One way around is to change the title to `Recent Improvements`?

> 3. [Add Support for `system` `ssl_trusted_certificate`](https://github.com/apache/apisix/pull/11809) (Contributor: [Revolyssup](https://github.com/Revolyssup))

Reviewer: ```suggestion
3. [Support the configuration of `system` in `ssl_trusted_certificate`](https://github.com/apache/apisix/pull/11809) (Contributor: [Revolyssup](https://github.com/Revolyssup))
```

cc @Revolyssup to x-check technical fact.

> With this feature, users can set `system` as the value to use when ca_certs are to be used from the system-defined `ca_certs`.

Reviewer: ```suggestion
With this feature, users can set `system` as the value when configuring `ssl_trusted_certificate`, to use the system-defined CA certificated.
```

cc @Revolyssup to x-check technical fact.

> 4. [Allow Workflow Configuration without `case`](https://github.com/apache/apisix/pull/11787) (Contributor: [Revolyssup](https://github.com/Revolyssup))

Reviewer: ditto, `workflow`?
